### ISSUE TITLE
Priority 1 accessibility issue: Filter behaviour 

### ISSUE THREAD

**Issue by alicerichmond** [wellcomecollection/wellcomecollection.org#6564]
**What**

Changing the setting of any user interface component automatically causes a change of context without advising the user of the behaviour prior to use.

When a user selects a filter, for example, a ‘Formats’ checkbox, the page automatically updates without informing users to this behaviour (please refer to Issue ID: DAC_Status_Update_01).

Likewise, the ‘Dates’ filters are applied dynamically as a user enters a date. However, they are not similarly removed when a user deletes the values; users must submit the form, however, there is no explicit submit button so users must press the ‘Enter’ key. There are no instructions to describe this behaviour.

Additionally, the ‘Date’ can be deleted using the ‘remove’ links on the page; however, this does not clear the form field date values within the collapsed panel. When a user then selects a ‘Formats’ filter, all of the filters are then re-applied to the page – including the ‘Dates’ filters which were previously cleared but whose values are still maintained by the Date input fields. This means users may end up applying filters unintentionally.

<img width="578" alt="Screenshot 2021-05-27 at 16 55 04" src="https://user-images.githubusercontent.com/7315126/119858344-4b5f9480-bf0c-11eb-9528-fdff22c1eb5f.png">

Solution:
An explicit submit button is recommended so that a user has a chance to review their selection before an action is triggered.

**Where**

Search boxes

**Comment by gestchild** [comment]
@alicerichmond @DominiqueMarshall @GarethOrmerod this is going to need your input.

The suggested solution is to only update search results when a user activates the submit button. However, we intentionally made the decision to auto update results when filters are applied/removed.

If we want to keep this behaviour, which I presume we do, another solution is to [inform the user of this behaviour before it happens](https://www.w3.org/WAI/WCAG21/Techniques/general/G13.html).

That leaves the questions of what we should say and where would the information should be placed.

**Comment by pollecuttn** [comment]
@GarethOrmerod and @DominiqueMarshall to discuss direction needed

**Comment by GarethOrmerod** [comment]
I think the problem is our choice of using a [listbox](https://www.w3.org/TR/wai-aria-practices-1.1/#Listbox) for that type of functionality.

> the interaction model conveyed by the listbox role to assistive technologies does not support interacting with elements inside of an option. Because of these traits of the listbox widget, it does not provide an accessible way to present a list of interactive elements, such as links, buttons, or checkboxes. To present a list of interactive elements, see the grid pattern.

The recommendation is use of a [grid pattern](https://www.w3.org/TR/wai-aria-practices-1.1/#grid) for what we are trying to acheive.

We do this for `More filters` so is the question — why are all the filters displayed not using a modal or [disclosure button](https://www.w3.org/TR/wai-aria-practices-1.1/#disclosure) to display the options?

**Comment by jennpb** [comment]
@AMBacelar This issue may be relevant to your work now. Let us know if your work will help solve this identified accessibility issue. Thank you!

**Comment by jennpb** [comment]
Just poking at this issue a bit.
I looked at the second of Gareth's zeplin links above (https://zpl.io/adydGXE) showing the single column filters overflow modal. The result count for each of the attributes updates dynamically when one is selected, so if we wanted to implement this pattern we'd sacrifice dynamic updates of the results count. 

If what we need to address the accessibility issue is a facet-by-facet explicit confirmation, we'd need to split out all of the 'more filters' into unique filters, and work out how to address the display of those filters.

Next steps:
- Discuss feasibility with @DominiqueMarshall and Devs

**Comment by rcantin-w** [comment]
Interesting, I don't think we'd seen these before. FWIW and as context, I believe that in Search, the decision was made to keep Filters changes as simple as possible as we weren't in a position to rediscover them with a designer extensively.

I'm inserting myself and @AMBacelar in as I also just found out there was visually buggy behaviour on our search filters on small desktops and having them on one column might be the fix we're looking for. Can I confirm these designs are what we're working towards?

**Comment by jennpb** [comment]
@rcantin-w No, I don't think they are what we're aiming toward - at least not without discussing it more widely!
